### Set the job name
#PBS -N bedtools

### Run in the queue named "long8gb"
#PBS -q short8gb

### Use the bourne again shell
#PBS -S /bin/bash

### Pass enviroment variables to the job
#PBS -V

### To send email when the job is completed:
#PBS -m ae
#PBS -M lynn.sanford@colorado.edu

### Specify localhost and an NFS filesystem to prevent file copy errors.
#PBS -e localhost:/Users/lysa8537/2016_07_RNASeq/05_bedtools_count/OUTERR/bedtools.count.gene_KCl_only_Rep1.err
#PBS -o localhost:/Users/lysa8537/2016_07_RNASeq/05_bedtools_count/OUTERR/bedtools.count.gene_KCl_only_Rep1.out

### Specify the number of cpus for your job
#PBS -l nodes=1:ppn=1

### Tell PBS how much memory you expect to use. Use units of 'b','kb', 'mb' or 'gb'.
#PBS -l mem=25gb

### Tell PBS the anticipated run-time for your job, where walltime=HH:MM:SS
#PBS -l walltime=20:00:00

### Switch to the working directory; by default TORQUE launches processes
### from your home directory.
cd $PBS_O_WORKDIR
echo Working directory is $PBS_O_WORKDIR

# Calculate the number of processors allocated to this run.
NPROCS=`wc -l < $PBS_NODEFILE`

# Calculate the number of nodes allocated.
NNODES=`uniq $PBS_NODEFILE | wc -l`

### Display the job context
echo Running on host `hostname`
echo Time is `date`
echo Directory is `pwd`
echo Using ${NPROCS} processors across ${NNODES} nodes

### Assign path variables 
SCRATCH="/scratch/Users/lysa8537"

### Load modules
module load bedtools2_2.22.0

### Run bedtools on sorted BAM files in a strand-specific, first position 0 manner
bedtools multicov -bams $SCRATCH/2016_07_RNASeq/KCl_Rep1/mapping/KCl_Rep1_trim_beg10_very_sensitive.sorted.bam -bed $HOME/2016_07_RNASeq/Annotations/rn6_gene_annotation.bed > $SCRATCH/2016_07_RNASeq/Count_files/KCl_only_Rep1.count.gene.txt

### Sync back from scratch
rsync $SCRATCH/2016_07_RNASeq/Count_files/KCl_only_Rep1.count.gene.txt $HOME/2016_07_RNASeq/05_bedtools_count/KCl_only/

#Final time stamp
echo Time is `date`
